pennylane_orquestra/orquestra_device.py

```diff
@@ -0,0 +1,378 @@
+
+        Returns:
+            str: string representation of terms making up the observable
+        """
+        if not self.analytic:
+            obs_wires = observable.wires
+            wires = self.wires.indices(obs_wires)
+            op_str = self.pauliz_operator_string(wires)
+        else:
+            op_str = self.qubit_operator_string(observable)
+
+        return op_str
+
+    @staticmethod
+    def pauliz_operator_string(wires):
+        """Creates an OpenFermion operator string based on the related wires
+        that can be passed when creating an ``openfermion.IsingOperator``.
+
+        This method is used if rotations are needed for the backend specified.
+        In such a case a string that represents measuring PauliZ on each of the
+        affected wires is used.
+
+        **Example**
+
+        >>> dev = QeQiskitDevice(wires=2)
+        >>> wires = [0, 1, 2]
+        >>> op_str = dev.pauliz_operator_string(wires)
+        >>> print(op_str)
+        [Z0 Z1 Z2]
+        >>> print(openfermion.IsingOperator(op_str))
+        1.0 [Z0 Z1 Z2]
+
+        Args:
+            wires (Wires): the wires the observable of the quantum function
+                acts on
+
+        Returns:
+            str: the ``openfermion.IsingOperator`` string representation
+        """
+        op_wires_but_last = [f"Z{w} " for w in wires[:-1]]
+
+        # No space after the last wire
+        op_last_wire = f"Z{wires[-1]}"
+        op_str = "".join(["[", *op_wires_but_last, op_last_wire, "]"])
+        return op_str
+
+    def qubit_operator_string(self, observable):
+        """Creates an OpenFermion operator string from an observable that can
+        be passed when creating an ``openfermion.QubitOperator``.
+
+        This method decomposes an observable into a sum of Pauli terms and
+        identities, if needed.
+
+        **Example**
+
+        >>> dev = QeQiskitDevice(wires=2)
+        >>> obs = qml.PauliZ(0)
+        >>> op_str = dev.qubit_operator_string(obs)
+        >>> print(op_str)
+        1 [Z0]
+        >>> obs = qml.Hadamard(0)
+        >>> op_str = dev.qubit_operator_string(obs)
+        >>> print(op_str)
+        0.7071067811865475 [X0] + 0.7071067811865475 [Z0]
+
+        Args:
+            observable (pennylane.operation.Observable): the observable to serialize
+
+        Returns:
+            str: the ``openfermion.QubitOperator`` string representation
+        """
+        accepted_obs = {"PauliX", "PauliY", "PauliZ", "Identity"}
+
+        if isinstance(observable, Tensor):
+            need_decomposition = any(o.name not in accepted_obs for o in observable.obs)
+        else:
+            need_decomposition = observable.name not in accepted_obs
+
+        if need_decomposition:
+            original_observable = observable
+
+            # Decompose the matrix of the observable
+            # This removes information about the wire labels used and
+            # consecutive integer wires are used
```

Comment:
@antalszava is this a bug in the `decompose_hamiltonian` function that we need to fix in PL core?

Reply:
Not a bug per se, it's just that `decompose_hamiltonian` takes a matrix and not a PennyLane `Observable`. So when obtaining the matrix for the observable, wire information in lost by default. Perhaps we could think about changing it or creating a variant that can take an observable and preserve wiring internally.